hash.c

```diff
@@ -3576,21 +3576,22 @@ rb_hash_to_h_block(VALUE hash)
 
 /*
  *  call-seq:
- *    to_h -> self or new_hash
  *    to_h {|key, value| ... } -> new_hash
+ *    to_h -> self or new_hash
  *
- *  For an instance of +Hash+, returns +self+.
- *
- *  For a subclass of +Hash+, returns a new +Hash+
- *  containing the content of +self+.
+ *  With a block given, returns a new hash whose content is based on the block;
+ *  the block is called with each entry's key and value;
+ *  it block should return a 2-element array
```

Comment:
Did you mean "the" here?

```suggestion
 *  the block should return a 2-element array
```